### PR TITLE
Added promise support for library

### PR DESCRIPTION
**Pull request by alexbjorlig**
Based on this excellent SO answer:
https://stackoverflow.com/a/36838115/3694288

The objective is to make the library more available and easy to use.

I was thinking about adding documentation in the README.md, bug I think most people would simply expect the promise nature of libraries these days. Maybe in the future we can have a talk about typescript support, giving end-users a better idea about the API of the library.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
-var stream = require('./stream');
-var es = require('event-stream');
-var _ = require('lodash');
+var stream = require("./stream");
+var es = require("event-stream");
+var _ = require("lodash");
```
Comment (addaleax) [inline lib/index.js:3]: Fwiw, these changes are not really in line with our common coding style, even if it’s not reported as a linting error currently

**Review by addaleax — APPROVED** [review]

File: lib/index.js (hunk 1 above)
Comment (alexbjorlig) [reply lib/index.js:3]: Ahh sorry - I think it was caused automatically when I saved in VS-code. Would you like me to fix and maybe add a `.vscode` setting with preference for `'` quotes - or maybe even adding a `prettier.config.js`if you have one from other projects that would fit?

Comment (alexbjorlig) [reply lib/index.js:3]: Ahh ok. Seems adding this to `package.json` makes eslint run: `"lint": "eslint lib test examples --fix"`. Do you wan't me to add that in a new commit?

Comment (addaleax) [reply lib/index.js:3]: sure, whichever way is easiest for you, as long as it works in the end :)

Comment (alexbjorlig) [reply lib/index.js]: Ok - I fixed the linting issues and added a `lint` script to fix it in the future.

Comment (alexbjorlig) [reply lib/index.js]: Is there something else I should add before the PR can be merged?

Comment (alexbjorlig) [reply lib/index.js]: Is there something else I should add before the PR can be merged?

**Comment by alexbjorlig** [comment]
Hi @addaleax . For some reason I can't mention you in the comment's above ⬆️

Are there other changes you wan't me to fix before merging? Otherwise I would say the PR is ready ☺️

Regarding the lodash issue, I will try to create a few PR's to simply remove lodash instead of replacing with the new version. Should be possible to accomplish everything with modern js.

**Comment by addaleax** [comment]
Yeah, I think this is just fine :+1: Feel free to keep changes coming here.

**Pull request merged**